services/database.js

```diff
@@ -0,0 +1,20 @@
+const Url = require("../models/url");
+const shortid = require("shortid");
+
+const findUrl = async (originalUrl) => {
+  return await Url.findOne({ originalUrl });
+};
+
+const createAndSaveUrl = async (originalUrl) => {
```

Comment:
Before a `url` is created and saved it should be validated first, If it's not, an error is thrown with a message indicating that the URL is invalid.

Reply:
Errors are not handled here and error handling to the database operations can help catch and handle any errors that might occur.

The refactored code to include error handling can be like this

```
const createAndSaveUrl = async (originalUrl) => {
  try {
    const shortUrl = shortid.generate();

    //Creating a new Url document with the originalUrl and shortUrl
    const url = new Url({
      originalUrl,
      shortUrl,
    });
    await url.save();
    return url;
  } catch (err) {
    console.error(`Error occurred while creating and saving URL`);
    throw err;
  }
};
```